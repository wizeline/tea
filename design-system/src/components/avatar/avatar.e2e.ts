import { newE2EPage } from '@stencil/core/testing';

describe('wz-avatar', () => {
  it('should render an avatar-component', async () => {
    const page = await newE2EPage();
    await page.setContent(
      `<wz-avatar xlarge square src="https://api.adorable.io/avatars/285/abott@adorable.png" />`,
    );
    const avatar = await page.find('wz-avatar');
    expect(avatar.outerHTML).toMatchSnapshot();
  });
});
