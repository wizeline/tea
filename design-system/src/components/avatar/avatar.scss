wz-avatar {
  div {
    background-size: cover;

    &.xsmall {
      width: 24px;
      height: 24px;
    }

    &.small {
      width: 32px;
      height: 32px;
    }

    &.medium {
      width: 64px;
      height: 64px;
    }

    &.large {
      width: 100px;
      height: 100px;
    }

    &.xlarge {
      width: 132px;
      height: 132px;
    }

    &.xxlarge {
      width: 164px;
      height: 164px;
    }

    &.round {
      border-radius: 50%;
    }

    &.square {
      border-radius: 4px;
    }
  }
}
