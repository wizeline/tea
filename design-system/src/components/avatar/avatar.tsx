import { Component, Prop, h } from '@stencil/core';
import sizeMatcher, { MEDIUM_SIZE } from '../../matchers/size';
import shapeMatcher, { ROUND } from '../../matchers/shape';

@Component({
  tag: 'wz-avatar',
  styleUrl: 'avatar.scss',
  shadow: false,
})
export class Avatar {
  @Prop()
  src: string;
  @Prop()
  backgroundsize: string;
  @Prop()
  square: boolean;
  // Size props
  @Prop()
  xsmall: boolean;
  @Prop()
  small: boolean;
  @Prop()
  medium: boolean;
  @Prop()
  large: boolean;
  @Prop()
  xlarge: boolean;
  @Prop()
  xxlarge: boolean;

  getClass() {
    const size = sizeMatcher(this) || MEDIUM_SIZE;
    const shape = shapeMatcher(this) || ROUND;

    return [shape, size].join(' ');
  }

  render() {
    const componentClass = this.getClass();
    const style = {
      backgroundImage: `url('${this.src}')`,
      backgroundSize: this.backgroundsize,
    };

    return <div class={componentClass} style={style} />;
  }
}
