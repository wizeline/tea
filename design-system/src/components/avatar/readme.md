# wz-avatar



<!-- Auto Generated Below -->


## Properties

| Property         | Attribute        | Description | Type      | Default     |
| ---------------- | ---------------- | ----------- | --------- | ----------- |
| `backgroundsize` | `backgroundsize` |             | `string`  | `undefined` |
| `large`          | `large`          |             | `boolean` | `undefined` |
| `medium`         | `medium`         |             | `boolean` | `undefined` |
| `small`          | `small`          |             | `boolean` | `undefined` |
| `square`         | `square`         |             | `boolean` | `undefined` |
| `src`            | `src`            |             | `string`  | `undefined` |
| `xlarge`         | `xlarge`         |             | `boolean` | `undefined` |
| `xsmall`         | `xsmall`         |             | `boolean` | `undefined` |
| `xxlarge`        | `xxlarge`        |             | `boolean` | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
