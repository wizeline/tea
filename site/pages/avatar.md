### Circle Avatar
```html
showSource: true
---
    <wz-avatar xlarge circle src="https://api.adorable.io/avatars/285/yellow@adorable.io.png" />
```

### Square Avatar
```html
showSource: true
---
    <wz-avatar xlarge square src="https://api.adorable.io/avatars/285/abott@adorable.png" />
```

