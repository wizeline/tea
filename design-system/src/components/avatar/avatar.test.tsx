import { Avatar } from './avatar';
import { TestWindow } from '@stencil/core/testing';

describe('wz-avatar', () => {
  it('renders value', async () => {
    const window = new TestWindow();
    const element = await window.load({
      components: [Avatar],
      html: `<wz-avatar xlarge square src="https://api.adorable.io/avatars/285/abott@adorable.png" />`,
    });

    expect(element).toMatchSnapshot();
  });
});
