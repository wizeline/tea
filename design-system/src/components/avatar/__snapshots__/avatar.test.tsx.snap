// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wz-avatar renders value 1`] = `
<wz-avatar
  class="hydrated"
  square=""
  src="https://api.adorable.io/avatars/285/abott@adorable.png"
  xlarge=""
>
  
  <div
    class="square xlarge"
  />
</wz-avatar>
`;
